examples/nodejs/lambda-examples/cloudwatch-metrics/infrastructure/lib/momento-metrics-stack.ts

```diff
@@ -0,0 +1,310 @@
+            "secretsmanager:GetSecretValue",
+            "logs:PutLogEvents",
+            "logs:CreateLogStream",
+            "ecs:*"
+          ],
+          resources: ["*"],
+        }));
+        taskDefinition.addToExecutionRolePolicy(new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "secretsmanager:ListSecrets",
+            "secretsmanager:GetSecretValue",
+            "logs:PutLogEvents",
+            "logs:CreateLogStream",
+            "ecs:*"
+          ],
+          resources: ["*"],
+        }));
+
+        new ecs.FargateService(this, 'MomentoMetricsECSFargateService', {
+          cluster,
+          taskDefinition,
+        });
+        break;
+      }
+      case exampleApp.Custom: {
+        console.log('Skipping to dashboard creation');
+        break;
+      }
+      default: {
+        throw new Error('Unimplemented CDK stack application');
+      }
+    }
+
+    logGroup.addMetricFilter('ExampleMetricFilterDuration', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Duration (Latency)',
+      filterPattern: FilterPattern.literal('{$.duration > 0}'),
+      metricValue: '$.duration',
+      unit: Unit.MILLISECONDS,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterRequestSize', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Request Size (bytes)',
+      filterPattern: FilterPattern.literal('{$.requestSize >= 0}'),
+      metricValue: '$.requestSize',
+      unit: Unit.BYTES,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterResponseSize', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Response Size (bytes)',
+      filterPattern: FilterPattern.literal('{$.responseSize >= 0}'),
+      metricValue: '$.responseSize',
+      unit: Unit.BYTES,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterGrpcStatusCode', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'GRPC Status Code',
+      filterPattern: FilterPattern.literal('{ $.status >= 0 }'),
+      metricValue: '$.status',
+    });
+
+    const dashboard = new Dashboard(this, 'MomentoMetricsCDKExampleDashboard', {
+      dashboardName: 'MomentoMetricsCDKExampleDashboard',
+      defaultInterval: cdk.Duration.hours(1),
+    });
+
+    const latency = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'Duration (Latency)',
+          namespace: 'MomentoMetricsCDKExample',
+          region: 'us-west-2',
+        }),
+      ],
+      stacked: false,
+      height: 6,
+      width: 6,
+      region: 'us-west-2',
+      title: 'Duration (Latency)',
+    });
+    latency.position(0, 0);
+
+    const errorCodes = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'GRPC Status Code',
+          namespace: 'MomentoMetricsCDKExample',
+          region: 'us-west-2',
+        }),
+      ],
+      stacked: false,
+      height: 6,
+      width: 6,
```

Comment:
I suspect that some of the layout properties you are setting on some of these widgets may align with the defaults.

In the past I have found that it's easier to manage programmatic dashboards if you keep the number of layout properties that you explicitly specify to a minimum. You will definitely need some height/width/position attributes, and you may also want to keep some of your annotation stuff. Heck, maybe you need everything here :) but just wanted to call out - i would experiment with omitting anything you don't think is absolutely necessary and see if it still looks okay, and if so, omit what you can.

totally your call on which things clear the bar of being necessary though!